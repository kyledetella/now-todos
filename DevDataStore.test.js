const devDataStore = require("./DevDataStore");

describe("DevDataStore", () => {
  beforeAll(() => {
    jest
      .spyOn(devDataStore, "__generateId")
      .mockImplementation(() => "test-id");
  });

  describe("createTodo", () => {
    it("adds new todo to collection", () => {
      devDataStore.createTodo("Foo");

      expect(devDataStore).toMatchSnapshot();
    });
  });

  describe("getTodos", () => {
    it("returns Todo collection", () => {
      expect(devDataStore.getTodos()).toMatchSnapshot();
    });
  });
});
