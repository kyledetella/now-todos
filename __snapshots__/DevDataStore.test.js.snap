// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevDataStore createTodo adds new todo to collection 1`] = `
DevDataStore {
  "__generateId": [MockFunction] {
    "calls": Array [
      Array [],
    ],
  },
  "_todos": Array [
    Object {
      "description": "Foo",
      "id": "test-id",
    },
  ],
}
`;

exports[`DevDataStore getTodos returns Todo collection 1`] = `
Array [
  Object {
    "description": "Foo",
    "id": "test-id",
  },
]
`;
